src/prettytable/prettytable.py

```diff
@@ -1087,6 +1087,27 @@ def add_column(self, fieldname, column, align="c", valign="t"):
                 f"{len(self._rows)}"
             )
 
+    def add_autoindex(self, align="c", valign="t", fieldname="Index"):
+
+        """Add an index column to the table.
+
+        Arguments:
+
+        align - desired alignment for this column - "l" for left, "c" for centre and
+            "r" for right
+        valign - desired vertical alignment for new columns - "t" for top,
+            "m" for middle and "b" for bottom"""
+
+        self._validate_align(align)
+        self._validate_valign(valign)
+        self._field_names.insert(0, fieldname)
+        self._align[fieldname] = align
+        self._valign[fieldname] = valign
+        c = 1
+        for i in range(0, len(self._rows)):
+            self._rows[i].insert(0, c)
+            c += 1
```

Comment:
How about using the existing values of `self.align` and `self.valign`?

For example:

```suggestion
    def add_autoindex(self, fieldname="Index"):

        """Add an auto-incrementing index column to the table.

        Arguments:

        fieldname - name of the field to contain the new column of data"""

        self._field_names.insert(0, fieldname)
        self._align[fieldname] = self.align
        self._valign[fieldname] = self.valign
        for i, row in enumerate(self._rows):
            row.insert(0, i + 1)
```

This also simplifies the loop a bit too.

The tests will need updating too.

Reply:
I installed the latest development version and it tracked some other changes that were made in the meantime, apart from mine, I'm sorry for the inconvenience. Do you think we should also add testing for different valign and align values of the table or is it not needed?

Reply:
I don't think we need to test different valign and align, should be enough they're tested elsewhere. Thanks!

Reply:
Thank you!